### DISCUSSION TITLE
Cookie Consent

### DISCUSSION THREAD

**Discussion by JJGrootveld** [nl-design-system/backlog#359] [Component suggestions]
<img width="766" alt="isuele weergave van een cookie consent component" src="https://github.com/user-attachments/assets/89f04331-bf8f-4af1-9575-5e63e0059e21">

&nbsp;

We willen graag meer leren over het Cookie Consent component. In welke vorm, of vormen, wordt deze momenteel ingezet en toegepast binnen jouw organisatie?

Deel hieronder een link naar Figma, Storybook of screenshots. Met deze informatie kunnen we bewijzen dat het component onderdeel moet worden van de NL Design System component library.

Daarnaast wordt het delen van onderzoek gewaardeerd. Dus heb je zelf onderzoek gedaan en zijn er inzichten voor dit component? Dan horen we dit ook graag!

## Naam

Cookie Consent Banner

## Beschrijving

Toont een venster over de pagina om cookies te accepteren of te weigeren die niet essentieel zijn voor het functioneren van de website of dienst.

## Ook bekend als

Cookie banner, Cookie consent banner

## Richtlijnen
- [Robert Roose: "Volgens mij is de richtlijn dat je geen cookies plaatst die ervoor zorgen dat je een cookie banner moet te tonen. Zoals bij rvo.nl, overheid.nl, belastingdienst.nl en rijksoverheid.nl"](https://codefornl.slack.com/archives/C7KDQRXMZ/p1732885227774019?thread_ts=1732876456.424239&cid=C7KDQRXMZ)

### Compliance
- https://www.autoriteitpersoonsgegevens.nl/themas/internet-slimme-apparaten/cookies/cookiewall

### Component
- https://github.com/porscheofficial/cookie-consent-banner#arrows_counterclockwise-consent-flow
- https://design-system.service.gov.uk/components/cookie-banner/
- https://www.duetds.com/components/cookie-consent/
- https://ds.blocks.gov.ie/components/library/cookie-banner/
- https://pie.design/patterns/cookie-banner/
- https://hds.hel.fi/components/cookie-consent/

### Pattern
- https://design-system.service.gov.uk/patterns/cookies-page/
- https://hds.hel.fi/patterns/cookies/basics/

**Comment by JJGrootveld** [comment]
## KVK
[kvk.nl](kvk.nl) (als je de cookie consent niet ziet -> clear cookies en refresh om te zien)

### Screenshots / Varianten

#### Screen size: 720px>

<img width="766" alt="visuele weergave van de cookie consent dialog op kvk.nl" src="https://github.com/user-attachments/assets/66a9ea7b-4d2e-4292-886e-08a05bedd325">
<img width="766" alt="visuele weergave van de cookie consent dialog op kvk.nl met een opengeklapte accordion met informatie over functionele cookies" src="https://github.com/user-attachments/assets/47ff86cf-bda2-4625-b3d4-89af3c5f230b">

#### Screen size: <719px

<img width="400" alt="visuele weergave van de cookie consent dialog op kvk.nl op een klein scherm van 375px breed" src="https://github.com/user-attachments/assets/1aee6ee6-4aa9-4494-9143-68dcfe09cd14">

### Notities
- nog aanvullen...

**Comment by RenateRoke** [comment]
## Informatie uit de Kennisbank digitoegankelijk voor toegankelijke cookiebanners

Met cookies kan je informatie verzamelen over het bezoek of over de bezoeker van je website. Deze informatie gebruik je voor functies op de website, voor marketing of inzicht in bezoekersaantallen.


De algemene verordening gegevensbescherming (AVG) zegt dat je bezoekers om toestemming moet vragen als je met cookies persoonsgegevens verwerkt. Je doet dit met een cookiemelding of cookie banner. Maak de cookiemelding toegankelijk, zodat iedereen zelf kiest welke gegevens wel en niet worden bewaard met cookies.

### Positie van de cookiemelding
Plaats de cookiemelding in code helemaal bovenaan de pagina. Net na de `<body>` en voor de skiplink. Zo komen bezoekers de cookiemelding als eerste tegen als ze het toetsenbord of een schermlezer gebruiken. Dit voorkomt dat mensen met een visuele beperking de cookiemelding te laat of helemaal niet tegenkomen.

### Cookiemelding van een leverancier
Er zijn veel leveranciers die het cookiebeheer en de melding voor je regelen. Dit kan heel makkelijk zijn, maar het kan ook zorgen voor een volledig ontoegankelijke website. Bijvoorbeeld wanneer je bezoeker via het toetsenbord vast komt te zitten in de cookiemelding.

Vraag naar een onderzoek dat digitale toegankelijkheid van het product aantoont of laat de leverancier een toegankelijkheidsonderzoek doen van de cookiemelding.

### Doe dit voor toegankelijkheid

- Gebruik alleen bewezen toegankelijke oplossingen van leveranciers.
- Zorg dat in de focusvolgorde de cookiemelding eerst komt.
- Maak alle onderdelen ook toegankelijk voor het toetsenbord.
- Pas op met sticky cookiemeldingen. Zoom je in bij cookmeldingen met een vaste plaats, dan verbergen ze vaak inhoud op de pagina.
- Zorg voor voldoende kleurcontrast.
- Zorg voor tekstalternatieven, bijvoorbeeld bij iconen.
- Maak teksten toegankelijk met koppen en eventueel lijsten.
- Geef formulierelementen een duidelijk label en een toegankelijke naam. Bijvoorbeeld bij selectievakken.
- Stel de taal van de cookiemelding goed in.
- Zorg dat je de cookiemelding makkelijk kan sluiten, ook met een toetsenbord.

**Comment by julezrulez** [comment]
## Voorbeelden cookiemeldingen
Ik onderzoek de cookiemeldingen al een tijdje maar heb dit onderzoek nog niet afgerond. Het onderzoek omvat ruim 450 hoofdwebsites van onder andere gemeenten, waterschappen, provincies en veiligheidsregio's. Ongeveer de helft van deze organisaties hebben een cookiemelding op de website. De ontwikkeling van het onderzoeksrapport is te volgen op https://www.digimonitor.nl/cookiemeldingen/
De zaken die voor deze discussie belangrijk zijn publiceer ik hier.

### Cookiemelding
De cookiemelding is er in de volgende smaken:
- cookiebalk, zie hieronder voorbeelden
- cookiepopup
- cookieknop

Al deze vormen zijn sticky of scrollen mee met de pagina. Bij de popup is vaak een actie vereist. Bij de cookiebalk kan je vaak de website al goed bekijken.

### Keuzemogelijkheid bij cookiemeldingen
Een cookiemelding biedt ongeacht de vorm (balk of popup) een of meerdere mogelijkheden aan:
- Sluit melding
- Accepteer cookies
- Accepteer noodzakelijke cookies
- Weiger cookies
- Wijzig cookie-instellingen
- Informatie over cookies
Voor cookie-instellingen wordt vaak een aparte pagina aangemaakt. Bij popups komt het voor dat de instellingen binnen de popup aangepast kunnen worden.

### Voorbeelden cookiemelding in een balk
#### smalle balk boven
https://www.montfoort.nl/
<img width="944" height="647" alt="Gemeente Montfoort" src="https://github.com/user-attachments/assets/db5aac5a-2049-4b2c-a3a5-91707ee38e19" />

https://ggd.groningen.nl/
<img width="944" height="647" alt="GGD Groningen" src="https://github.com/user-attachments/assets/34f0ef35-4855-4817-a3d4-b6e0791fc157" />

https://www.provinciegroningen.nl/
<img width="944" height="647" alt="Provincie Groningen" src="https://github.com/user-attachments/assets/02a8bbd3-353a-4956-87b8-dcba53fd1dfe" />

https://www.loonopzand.nl/
<img width="944" height="647" alt="Gemeente Loon op Zand" src="https://github.com/user-attachments/assets/ae73a1e1-a33a-4156-b35a-87fdc61ffa63" />

https://www.vechtstromen.nl/
<img width="948" height="650" alt="Waterschap Vechtstromen" src="https://github.com/user-attachments/assets/9274796a-aa4c-4604-adb9-b45614c0768a" />

#### brede balk boven
https://www.beekdaelen.nl/x
<img width="944" height="647" alt="Gemeente Beekdaelen" src="https://github.com/user-attachments/assets/234f9a50-f585-4039-8206-2927eff14a2c" />

https://www.dewolden.nl/
<img width="944" height="647" alt="Gemeente De Wolden" src="https://github.com/user-attachments/assets/f00f33bf-28fe-4f58-8637-919d9c127ebd" />

https://www.middendelfland.nl/
<img width="944" height="647" alt="Gemeente Midden-Delfland" src="https://github.com/user-attachments/assets/3bccb953-c158-40cd-822f-72e8e70e15e9" />

https://www.kerkrade.nl/home
<img width="948" height="649" alt="Gemeente Kerkrade" src="https://github.com/user-attachments/assets/bb17cc31-6a29-473a-b04d-ce5faec73f95" />

https://www.vrnhn.nl/
<img width="944" height="647" alt="Veiligheidsregio Noord-Holland Noord" src="https://github.com/user-attachments/assets/095ad62a-29d6-49bf-85bd-79006b9ff38d" />

#### Sticky banners
Deze komen we bijvoorbeeld tegen op de volgende websites.
Allereerst sticky bovenaan de pagina:
- https://www.oss.nl/
- https://www.sliedrecht.nl/
- https://www.limburg.nl/
- https://www.waterschaplimburg.nl/

en sticky onderaan de pagina:
- https://www.zuid-holland.nl/
- https://veiligheidsregioaa.nl/
- https://www.hdsr.nl/
- https://www.ggdrotterdamrijnmond.nl/

**Comment by julezrulez** [comment]
Op basis  van https://www.autoriteitpersoonsgegevens.nl/themas/internet-slimme-apparaten/cookies/heldere-cookiebanners  kun je bepalen of je op correcte wijze om toestemming vraagt. Bepalend is natuurlijk het soort cookies dat je opslaat. Van de volgende voorbeelden is nog niet duidelijk of het uiterlijk van de buttons en links op deze wijze gebruikt mag worden.

SITUATIE 1: De cookies van derden staan wel uit, maar je moet zelf eerst de knop zien te vinden.
<img width="1278" height="221" alt="cookiemelding als knop tegen de onderrand van de browser aan" src="https://github.com/user-attachments/assets/6ca4c65b-19c1-4bd8-aa9a-7df02dd8ce22" />

SITUATIE 2: De cookiemelding wordt direct verborgen achter de feedbackfunctie.
<img width="956" height="493" alt="popup verschijnt, cookiemelding niet bereikbaar" src="https://github.com/user-attachments/assets/3d50f7d3-69f8-4c80-821e-5cbaa016809d" />

SITUATIE 3: De cookieverklaring verstopt achter een chatbot.
<img width="956" height="105" alt="Cookieknop verstopt" src="https://github.com/user-attachments/assets/ecc8cab4-adb7-43e2-9253-b2d544712d8a" />

SITUATIE 4: Is duidelijk genoeg dat de schuifjes uit staan (als je bijv. kleurenblind bent?)?
<img width="955" height="511" alt="checkboxes niet duidelijk" src="https://github.com/user-attachments/assets/2c528d6d-5876-4e5a-b7c4-7513bde21473" />

SITUATIE 5: Default aan en 2 knoppen "Accepteer" en "alles accepteren"
<img width="957" height="313" alt="checkboxes staan aan en 2 haast gelijknamige knoppen" src="https://github.com/user-attachments/assets/d64caa9a-ca63-428d-8023-f0f91695d2f1" />

SITUATIE 6: Alleen een Opslaan knop. Is dat voldoende?
<img width="960" height="442" alt="Alleen een Opslaan-knop" src="https://github.com/user-attachments/assets/73a32912-3746-49be-b494-efec13c22d0a" />

SITUATIE 7: Is op deze manier uit- en aanschakelen toegestaan?
<img width="962" height="118" alt="Akkoord en onduidelijke linkjes" src="https://github.com/user-attachments/assets/819bec72-d413-45f3-a7da-bfd8c5a7a6fc" />

SITUATIE 8: Dit moet je kunnen weigeren toch?
<img width="957" height="173" alt="Alleen Ik ga akkoord" src="https://github.com/user-attachments/assets/bc2044d8-fa64-4d36-841d-ce546879688c" />

SITUATIE 9: Dit moet je toch kunnen weigeren?
<img width="957" height="221" alt="Melding heeft alleen een ik ga akkoord-knop" src="https://github.com/user-attachments/assets/6ed6d317-d94e-422e-bdd0-1265736727fa" />

SITUATIE 10: Alleen aanpassen ziet er anders uit; dit mag toch?
<img width="957" height="286" alt="31-popup-links-cookieyes" src="https://github.com/user-attachments/assets/62390c63-90e8-47c1-9b35-461a1ef9ce00" />

**Comment by julezrulez** [comment]
<img width="986" height="597" alt="2 cookievensters Nederlands en Engels over elkaar heen" src="https://github.com/user-attachments/assets/beb0c72e-0c47-4c0b-afd4-76e98b799449" />

In een beeld gevangen. Dubbelop bestaat dus ook. Opvallend is de volgorde van de knoppen in beide vensters.
